src/java.base/share/classes/java/io/ObjectInputFilter.java

```diff
@@ -197,15 +360,35 @@
     }
 
     /**
-     * A utility class to set and get the system-wide filter or create a filter
-     * from a pattern string. If a system-wide filter is set, it will be
-     * used for each {@link ObjectInputStream} that does not set its own filter.
+     * A utility class to set and get the JVM-wide deserialization filter factory,
+     * the static JVM-wide filter, or to create a filter from a pattern string.
+     * If a JVM-wide filter factory or static JVM-wide filter is set, it will determine the filter
+     * to be used for each {@link ObjectInputStream}, and be combined with a
+     * stream-specific filter, if one is set.
+     * <p>
+     * When each {@link ObjectInputStream#ObjectInputStream() ObjectInputStream}
+     * is created the {@linkplain Config#getSerialFilterFactory() filter factory}
+     * is invoked to determine the initial filter for the stream. A stream-specific filter can be set with
```

Comment:
Maybe the concept of system-wide filter should be introduced first, then you could say that the factory is invoked with `null` as first parameter and the system-wide filter as second parameter when the `ObjectInputStream` is created. Then the next paragraph could expand on what happens when `setObjectInputFilter` is called.